WordPress/openverse
# Add 'revoked' field to ThrottledApplication to enable easily revoking access to client applications violating openverse TOS

## Fixes



Fixes #4321  by @sarayourfriend 

## Description


Add 'revoked' field to `ThrottledApplication` to enable easily revoking access to client applications violating openverse TOS.


## Testing Instructions

- Create a `ThrottledApplication` on Django admin and go ahead to use the generated credentials to authenticate and make request(s) to the API. This should work as expected and return results.

- Head back to your ThrottledApplication on Django admin, edit this application by checking the 'revoked' field and save. Try making requests to the API and you should get a `401` `PermissionDenied`.

Additionally, a unit test exists for this at `api/test/integration/test_auth.py::test_revoked_application_access`


## Checklist


      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
